arroyo-sql/src/expressions.rs

```diff
@@ -1195,7 +1202,77 @@ pub enum StringFunction {
     Rpad(Box<Expression>, Box<Expression>, Option<Box<Expression>>),
     Rtrim(Box<Expression>, Option<Box<Expression>>),
 }
+// postgres has to be active for build for some reason?
+
+#[derive(Debug)]
+pub enum HashFunction {
+    MD5,
+    SHA224,
+    SHA256,
+    SHA384,
+    SHA512,
+}
 
+impl ToString for HashFunction {
+    fn to_string(&self) -> String {
+        match self {
+            Self::MD5 => "md5".to_string(),
+            Self::SHA224 => "sha224".to_string(),
+            Self::SHA256 => "sha256".to_string(),
+            Self::SHA384 => "sha384".to_string(),
+            Self::SHA512 => "sha512".to_string(),
+        }
+    }
+}
+
+impl TryFrom<BuiltinScalarFunction> for HashFunction {
+    type Error = anyhow::Error;
+
+    fn try_from(value: BuiltinScalarFunction) -> Result<Self> {
+        match value {
+            BuiltinScalarFunction::MD5 => Ok(Self::MD5),
+            BuiltinScalarFunction::SHA224 => Ok(Self::SHA224),
+            BuiltinScalarFunction::SHA256 => Ok(Self::SHA256),
+            BuiltinScalarFunction::SHA384 => Ok(Self::SHA384),
+            BuiltinScalarFunction::SHA512 => Ok(Self::SHA512),
+            _ => bail!("function {} is not a hash function", value),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct HashExpression {
+    function: HashFunction,
+    input: Box<Expression>,
+}
+
+impl HashExpression {
+    pub fn new(function: BuiltinScalarFunction, input: Box<Expression>) -> Result<Expression> {
+        Ok(Expression::Hash(HashExpression {
+            function: function.try_into()?,
+            input,
+        }))
+    }
+}
+
+impl ExpressionGenerator for HashExpression {
+    fn to_syn_expression(&self) -> syn::Expr {
+        let input = self.input.to_syn_expression();
+        let hash_fn = format_ident!("{}", self.function.to_string());
+        parse_quote!({
```

Comment:
This is only correct if self.input.is_nullable(). Take a look at some of the other single-argument methods for how to handle the two cases.

Reply:
Should be better now with the check on `nullable()`